### DISCUSSION TITLE
How do I get the current logged user inside a Dialog.ts file?

### DISCUSSION THREAD

**Discussion by dBianchii** [serenity-is/Serenity#6264] [Help]
I need to get the current userId that is logged in. How can I do that?

**Comment by dBianchii** [comment]
I think I figured it out myself:

I sent an $ajax to an Actions Page:

```ts
                    var LoggeduserId: string;
                    $.ajax({
                        method: "post",
                        url: "/Portal/OrderActions/GetLoggedUserId",
                        success: function (data) {
                            LoggeduserId = data;
                        },
                        timeout: 200000
                    });
```

and in my ActionsPage.CS:

```cs
[HttpPost]
public ActionResult GetLoggedUserId()
{
   var userId = Serenity.Authorization.UserId;
   return Content(userId);
}
```

**Reply by VictorTomaili** [reply]
Check this and related references. 
https://github.com/serenity-is/Serene/blob/master/Serene/Serene.Web/Modules/Administration/User/Authentication/ScriptUserDefinition.cs

**Reply by dBianchii** [reply]
Much better. I can just add the line `result.LoggedUserId = Serenity.Authorization.UserId;` inside my UserEndpoint.cs file after delcaring `public String LoggedUserId { get; set; }` inside my ScriptUserDefinition.cs file.

**Reply by VictorTomaili** [reply]
add user Id [there](https://github.com/serenity-is/Serene/blob/master/Serene/Serene.Web/Modules/Administration/User/UserDataScript.cs#L48) and access it from the script with this `Authorization.userDefinition.UserId` from anywhere in typescript.
